lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp

```diff
@@ -767,13 +771,13 @@ void MayaUsdRPrim::_SyncSharedData(
     }
 
     if (HdChangeTracker::IsVisibilityDirty(*dirtyBits, id)) {
-        bool usdVisibility = delegate->GetVisible(id);
+        sharedData.visible = delegate->GetVisible(id) && _displayLayerModes._visibility;
 
         // Invisible rprims don't get calls to Sync or _PropagateDirtyBits while
         // they are invisible. This means that when a prim goes from visible to
         // invisible that we must update every repr, because if we switch reprs while
         // invisible we'll get no chance to update!
-        if (!usdVisibility)
+        if (!sharedData.visible)
```

Comment:
This fixes an issue with visibility when a display layer is switched into invisible state while in multi-viewport